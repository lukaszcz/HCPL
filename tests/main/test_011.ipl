
let f = \x . x + 7;

let g = \y . y + 7;

let q1 = quote (f 10);
let q2 = quote (g 10);

if q1 = q2 then
   print 1
else
   print 0;

if quote f = quote (\z . z + 7) then
   print 1
else
   print 0;

if quote f = quote (\z . z * 7) then
   print 1
else
   print 0;

match quote f with
| quote (\u { u + %x }) -> print x
| %_ -> print "zonk";

match quote g with
| quote (\u { u + %x }) -> print x
| %_ -> print "zonk";

match q1 with
| quote (\u { u + %x } 10) -> print x
| %_ -> print "zonk";

match q1 with
| quote (\u { u + %x } 1) -> print x
| %_ -> print "OK";

match q1 with
| q2 -> print "OK"
| %_ -> print "zonk"
