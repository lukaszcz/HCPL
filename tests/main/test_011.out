1
1
0
7
7
7
OK
OK
